lib/wmic.js

```diff
@@ -0,0 +1,99 @@
+var format = require('util').format
+var os = require('os')
+var spawn = require('child_process').spawn
+
+function wmic (pid, options, done) {
+  var prefix = 'ProcessId='
+  var whereClause = ''
+
+  if (!Array.isArray(pid)) {
+    pid = parseInt(pid, 10)
+    if (!options.history[pid]) options.history[pid] = {}
+    whereClause = prefix + pid
+  } else {
+    for (var i = 0; i < pid.length; i++) {
+      var curr = parseInt(pid[i], 10)
+      if (!options.history[curr]) options.history[curr] = {}
+      if (i === 0) {
+        whereClause += prefix + curr
+      } else {
+        whereClause += ' or ' + prefix + curr
+      }
+    }
+  }
+
+  // http://social.msdn.microsoft.com/Forums/en-US/469ec6b7-4727-4773-9dc7-6e3de40e87b8/cpu-usage-in-for-each-active-process-how-is-this-best-determined-and-implemented-in-an?forum=csharplanguage
+  var args = ['PROCESS', 'where', '"' + whereClause + '"', 'get', 'ProcessId,workingsetsize,usermodetime,kernelmodetime']
+  var wmic = spawn('wmic', args, {detached: true, windowsHide: true, windowsVerbatimArguments: true})
+  var stdout = ''
+  var stderr = ''
+
+  // Note: On Windows the returned value includes fractions of a second. Use Math.floor() to get whole seconds.
+  var uptime = Math.floor(os.uptime())
+
+  wmic.stdout.on('data', function (d) {
+    stdout += d.toString()
+  })
+
+  wmic.stderr.on('data', function (d) {
+    stderr += d.toString()
+  })
+
+  wmic.on('error', function (err) {
+    console.error('[pidusage] Command "wmic ' + args.join(' ') + '" failed with error %s', err)
```

Comment:
This should be returned into the callback.

Reply:
Close will be called after this.

Reply:
Yes but printing an error on the console is ugly!

Reply:
If the user needs that he can print using the error returned by the callback.